neuronpp/utils/record.py

```diff
@@ -143,17 +146,42 @@ def _plot_animate(self, steps=10000, y_lim=None, position=None):
 
                 ax.set_xlim(t.min(), t.max())
                 if y_lim is None:
-                    ax.set_ylim(r.min()-(np.abs(r.min()*0.05)), r.max()+(np.abs(r.max()*0.05)))
+                    ax.set_ylim(r.min() - (np.abs(r.min() * 0.05)), r.max() + (np.abs(r.max() * 0.05)))
 
                 # update data
                 line.set_data(t, r)
-
+                # info draw triangles for true and predicted classes
+                true_x, true_y, pred_x, pred_y = self._class_tcks(label=i, true_class=true_class, pred_class=pred_class,
+                                                                  t=t, r=r, stepsize=stepsize, dt=dt)
+                ax.scatter(true_x, true_y, c="orange", marker="^", alpha=0.95)
+                ax.scatter(pred_x, pred_y, c="magenta", marker="v", alpha=0.95)
+
+            # info join plots by removing labels and ticks from subplots that are not on the edge
+            for key in self.axs:
+                for ax in self.axs[key]:
+                    ax[0].label_outer()
+            fig.subplots_adjust(left=0.09, bottom=0.075, right=0.99, top=0.98, wspace=None, hspace=0.00)
             fig.canvas.draw()
             fig.canvas.flush_events()
 
         if create_fig:
             plt.show(block=False)
 
+    def _class_tcks(self, label, true_class, pred_class, t, r, stepsize, dt):
+        n = len(true_class)
+        x = t[::int(2 * stepsize / dt)][-n:]
+        true_x = []
+        pred_x = []
+        for k in range(n):
+            # get the true classes for the current label
+            if true_class[k] == label:
+                true_x.append(x[k])
+            if pred_class[k] == label:
+                pred_x.append(x[k])
+        true_y = [min(r) + 2] * len(true_x)
+        pred_y = [max(r) - 2] * len(pred_x)
```

Comment:
Need to find better true_y (position of lower triangle) and red_y (position of upper one). All goes well as long as the motor neurone are activated.